### ISSUE TITLE
Feature Idea: Lookup column/data type

### ISSUE THREAD

**Issue by mblennegard** [mattermost/focalboard#340]
**Summary:**
Lookup column/data type to relate cards/items in different boards.
This lookup column should be clickable to open up the card in the other board directly.
There should also be the possibility to add additional columns from the other board to the board with the lookup column.

**How important this is to me and why:**
Medium

Use cases:
1. Ability to have a board with e.g. Clients where each card contains lots of client information. Then have a projects board where the client can be specified in a lookup column, with an easy link back to the card in the Client board.
2. Ability to add additional cards from e.g. the Clients board (e.g. Client contact person) to the projects board, therefore eliminating the need for having to open the entire Client card.

**Additional context / similar features:**
Similar to lookup columns in SharePoint.

This might be loosely related to issue #324 but when reading that issue I felt these two warranted their own issues.

**Comment by chenilim** [comment]
Thanks @mblennegard! I wasn't aware of lookup columns in SharePoint before (found [this article](https://sharepointmaven.com/how-to-create-a-lookup-column-in-sharepoint/) on it). Notion has a similar set of relational features. My gut opinion is these are advanced features we should think carefully about how to layer in. Too much and it becomes like managing a database (e.g. dealing with a network of foreign keys). Too little and data quickly gets out of sync.

Having the use cases helps a lot. My quick take is a link would be a light-weight workaround for this, but maybe a "lookup link" might be a way to make it easier to create that link. Have to think about this more. In the meantime, use cases from other folks would be helpful. Thanks!

**Comment by josh-hemphill** [comment]
Yep, I'd be fine having it just generate like an unfurled link. Right now I can't even link to cards in another board right now on desktop. I've got lots of mini-projects that don't make sense to have a board for each, so I have board to track the projects and board of tasks, that I'd love to be able to have link to their respective projects; so defining a property type as a lookup of the other board would be very useful.

**Comment by nickadamson** [comment]
A 'Relation' property ala Notion's would be fantastic. I could be misunderstanding this request, but it sounds like you're looking for a [relational database](https://www.oracle.com/database/what-is-a-relational-database/).

Here's a [quick overview](https://www.keepproductive.com/blog/relational-database-notion) of how you can use them in notion. 

Personally, I would be using this for daily tracking too. But it's also useful for CRM/Project, Team/Company, etc relations; anything where you might have multiple dbs with some duplicated data but multiple purposes.

@chenilim, I noticed [your comment](https://github.com/mattermost/focalboard/issues/324#issuecomment-1095222512) on #324.. is this at all similar to what you've got planned for linking cards?

**Comment by chenilim** [comment]
Right now, we're scoping out a "related cards" feature where you can list multiple other related cards in a card, e.g. for dependent cards (such as epics to tasks). Still in the planning phase, but we hope to put together a wireframe for more feedback soon. cc @asaadmahmood, @wuwinson.

The "workaround" for now is you can use a URL property to store a link to another card (just one though).

Right now, we don't foresee a full blown relational table type (like Notion or a DB), but it's really helpful to hear about the scenarios (such as CRM), which we might be able to at least partially address in a more lightweight way.
